src/rpm/headers/lead.rs

```diff
@@ -10,119 +10,50 @@ use crate::errors::*;
 /// Used to contain valid data, now only a very limited subset is used
 /// and the remaining data is set to fixed values such that compatibility is kept.
 /// Only the "magic number" is still relevant as it is used to detect rpm files.
-#[derive(Eq)]
+#[derive(BinRead, BinWrite, Eq, PartialEq, Debug)]
 pub struct Lead {
     magic: [u8; 4],
     major: u8,
     minor: u8,
     package_type: u16,
     arch: u16,
-    name: [u8; 66],
+    #[br(map = |v: [u8; 66]| String::from_utf8_lossy(&v).to_string())]
+    #[bw(map = |name: &String| {
+        let mut name_arr = [0u8; 66];
+        let name_size = std::cmp::min(name_arr.len() - 1, name.len());
+        name_arr[..name_size].clone_from_slice(&name.as_bytes()[..name_size]);
+        name_arr
+    })]
+    name: String,
```

Comment:
I would prefer to keep the data in the lead structure identical to what was on disk, rather than storing it as a potentially modified string